samples/nrf9160/modem_shell/src/gnss/gnss_socket.c

```diff
@@ -0,0 +1,947 @@
+		/* More than five lowest bits set, invalid bitmask */
+		shell_error(shell_global,
+			    "GNSS: Invalid NMEA bitmask 0x%x", mask);
+		return -EINVAL;
+	}
+
+	nmea_mask = mask;
+
+	return 0;
+}
+
+int gnss_set_priority_time_windows(bool value)
+{
+	int err;
+	int opt;
+
+	if (value) {
+		opt = NRF_SO_GNSS_ENABLE_PRIORITY;
+	} else {
+		opt = NRF_SO_GNSS_DISABLE_PRIORITY;
+	}
+
+	err = nrf_setsockopt(fd, NRF_SOL_GNSS, opt, NULL, 0);
+	if (err) {
+		shell_error(shell_global,
+			    "GNSS: Failed to set priority time windows");
+	}
+
+	return err;
+}
+
+int gnss_set_agps_data_enabled(bool ephe, bool alm, bool utc, bool klob,
+			       bool neq, bool time, bool pos, bool integrity)
+{
+#if defined(CONFIG_SUPL_CLIENT_LIB)
+	agps_inject_ephe = ephe;
+	agps_inject_alm = alm;
+	agps_inject_utc = utc;
+	agps_inject_klob = klob;
+	agps_inject_neq = neq;
+	agps_inject_time = time;
+	agps_inject_pos = pos;
+	agps_inject_int = integrity;
+
+	return 0;
+#else
+	shell_error(shell_global,
+		    "GNSS: Enable CONFIG_SUPL_CLIENT_LIB for AGPS support");
+	return -EOPNOTSUPP;
+#endif
+}
+
+int gnss_set_agps_automatic(bool value)
+{
+#if defined(CONFIG_SUPL_CLIENT_LIB)
+	agps_automatic = value;
+
+	return 0;
+#else
+	shell_error(shell_global,
+		    "GNSS: Enable CONFIG_SUPL_CLIENT_LIB for AGPS support");
+	return -EOPNOTSUPP;
+#endif
+}
+
+int gnss_inject_agps_data(void)
+{
+#if defined(CONFIG_SUPL_CLIENT_LIB)
+	gnss_init();
+
+	(void)memset(&agps_data, 0, sizeof(agps_data));
+	if (agps_inject_ephe) {
+		agps_data.sv_mask_ephe = 0xffffffff;
+	}
+	if (agps_inject_alm) {
+		agps_data.sv_mask_alm = 0xffffffff;
+	}
+	if (agps_inject_utc) {
+		agps_data.data_flags |= BIT(NRF_GNSS_AGPS_GPS_UTC_REQUEST);
+	}
+	if (agps_inject_klob) {
+		agps_data.data_flags |= BIT(NRF_GNSS_AGPS_KLOBUCHAR_REQUEST);
+	}
+	if (agps_inject_neq) {
+		agps_data.data_flags |= BIT(NRF_GNSS_AGPS_NEQUICK_REQUEST);
+	}
+	if (agps_inject_time) {
+		agps_data.data_flags |=
+			BIT(NRF_GNSS_AGPS_SYS_TIME_AND_SV_TOW_REQUEST);
+	}
+	if (agps_inject_pos) {
+		agps_data.data_flags |= BIT(NRF_GNSS_AGPS_POSITION_REQUEST);
+	}
+	if (agps_inject_int) {
+		agps_data.data_flags |= BIT(NRF_GNSS_AGPS_INTEGRITY_REQUEST);
+	}
+
+	k_work_submit_to_queue(&gnss_work_q, &get_agps_data_work);
+
+	return 0;
+#else
+	shell_error(shell_global,
+		    "GNSS: Enable CONFIG_SUPL_CLIENT_LIB for AGPS support");
+	return -EOPNOTSUPP;
+#endif
+}
+
+int gnss_set_pvt_output_level(uint8_t level)
+{
+	if (level < 0 || level > 2) {
+		return -EINVAL;
+	}
+
+	pvt_output_level = level;
+
+	return 0;
+}
+
+int gnss_set_nmea_output_level(uint8_t level)
+{
+	if (level < 0 || level > 1) {
+		return -EINVAL;
+	}
+
+	nmea_output_level = level;
+
+	return 0;
+}
+
+int gnss_set_event_output_level(uint8_t level)
+{
+	if (level < 0 || level > 1) {
+		return -EINVAL;
+	}
+
+	event_output_level = level;
+
+	return 0;
+}
+
+static void gnss_init(void)
+{
+	if (fd > -1) {
+		return;
+	}
+
+	k_work_q_start(&gnss_work_q, gnss_workq_stack_area,
```

Comment:
@tokangas Updating to new NCS master, compiler gives warning from this line:
```
./src/gnss/gnss_socket.c: In function 'gnss_init':
../src/gnss/gnss_socket.c:925:2: warning: 'k_work_q_start' is deprecated [-Wdeprecated-declarations]
  k_work_q_start(&gnss_work_q, gnss_workq_stack_area,
  ^~~~~~~~~~~~~~
In file included from /home/tor1/work/ncs/zephyr/include/zephyr.h:18,
                 from ../src/gnss/gnss_socket.c:10:
/home/tor1/work/ncs/zephyr/include/kernel.h:3665:20: note: declared here
 static inline void k_work_q_start(struct k_work_q *work_q,
                    ^~~~~~~~~~~~~~

```

Reply:
Missed that one earlier, fixed now.